server/src/main/java/org/opensearch/plugins/PluginInfo.java

```diff
@@ -69,9 +69,33 @@
     private final Version opensearchVersion;
     private final String javaVersion;
     private final String classname;
-    private final String customFolderName;
     private final List<String> extendedPlugins;
     private final boolean hasNativeController;
+    private String customFolderName;
+
+    /**
+     * Construct plugin info.
+     *
+     * @param name                  the name of the plugin
+     * @param description           a description of the plugin
+     * @param version               an opaque version identifier for the plugin
+     * @param opensearchVersion     the version of OpenSearch the plugin was built for
+     * @param javaVersion           the version of Java the plugin was built with
+     * @param classname             the entry point to the plugin
+     * @param extendedPlugins       other plugins this plugin extends through SPI
+     * @param hasNativeController   whether or not the plugin has a native controller
+     */
+    public PluginInfo(String name, String description, String version, Version opensearchVersion, String javaVersion,
```

Comment:
Can we write this constructor in terms of the other one?

Reply:
Makes sense! Updated it.